File: apple/HDHROpenKit/Sources/HDHROpenKit/Utilities/Logger.swift
import Foundation
import os

public enum Log {
    private static let subsystem = "org.hdhropen.client"

    public static let general = os.Logger(subsystem: subsystem, category: "general")
    public static let network = os.Logger(subsystem: subsystem, category: "network")
    public static let auth = os.Logger(subsystem: subsystem, category: "auth")
    public static let player = os.Logger(subsystem: subsystem, category: "player")
    public static let dvr = os.Logger(subsystem: subsystem, category: "dvr")
}

File: android/core/src/main/kotlin/org/hdhropen/kit/utilities/Logger.kt
package org.hdhropen.kit.utilities

import android.util.Log as AndroidLog

class TaggedLogger(private val tag: String) {
    fun debug(message: String) {
        AndroidLog.d(tag, message)
    }

    fun info(message: String) {
        AndroidLog.i(tag, message)
    }

    fun warning(message: String) {
        AndroidLog.w(tag, message)
    }

    fun error(message: String, throwable: Throwable? = null) {
        if (throwable != null) {
            AndroidLog.e(tag, message, throwable)
        } else {
            AndroidLog.e(tag, message)
        }
    }
}

object Log {
    val general = TaggedLogger("HDHROpen.General")
    val network = TaggedLogger("HDHROpen.Network")
    val player = TaggedLogger("HDHROpen.Player")
    val dvr = TaggedLogger("HDHROpen.DVR")
    val auth = TaggedLogger("HDHROpen.Auth")
}
